From: Zebediah Figura <zfigura@codeweavers.com>
Subject: [PATCH] makefiles: Add libntdll.def to the dependencies for native Unix libraries.
Message-Id: <20210917224433.2051074-1-zfigura@codeweavers.com>
Date: Fri, 17 Sep 2021 17:44:33 -0500

These require libntdll.def to exist, otherwise -lntdll is not correctly
detected as a DLL with an associated Unix library by winegcc, and is
instead treated as a system dependency.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=51749
Signed-off-by: Zebediah Figura <zfigura@codeweavers.com>
---
 tools/makedep.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/makedep.c b/tools/makedep.c
index 59857762edd..b0091f0c461 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -3369,6 +3369,7 @@ static void output_module( struct makefile *make )
 
             strarray_add( &unix_imports, "ntdll" );
             strarray_add( &unix_deps, obj_dir_path( top_makefile, "dlls/ntdll/ntdll.so" ));
+            strarray_add( &unix_deps, obj_dir_path( top_makefile, "dlls/ntdll/libntdll.def" ));
             strarray_add( &unix_imports, "winecrt0" );
             if (spec_file) strarray_add( &unix_deps, spec_file );
 

-- 
2.33.0

